shell.nix

```diff
@@ -0,0 +1,31 @@
+# Licensed under the Apache License, Version 2.0 or the MIT License.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
```

Comment:
I don't know whether we've committed to a license yet. It might be fine to have this file be licensed under Apache/MIT (I'm guessing this is left over from basing this on the `shell.nix` in the Tock kernel repo), but given that this PR is basically just the bare minimum shell expression to get a Rust toolchain-shell from fenix, I think it's best to remove this for now.